src/main/java/org/jenkins_ci/plugins/flexible_publish/ConditionalPublisher.java

```diff
@@ -169,20 +170,41 @@ public ConditionalPublisherDescriptor getDescriptor() {
     }
 
     public boolean prebuild(final AbstractBuild<?, ?> build, final BuildListener listener) {
-        for (BuildStep publisher: getPublisherList()) {
-            if (!runner.prebuild(condition, publisher, build, listener)) {
-                return false;
-            }
+        if (getPublisherList().size() <= 0) {
+            return true;
+        } else if (getPublisherList().size() > 1) {
+            return runner.prebuild(condition, new RunAllBuilder(getPublisherList()), build, listener);
         }
-        return true;
+        return runner.prebuild(condition, getPublisherList().get(0), build, listener);
     }
 
     public boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher, final BuildListener listener)
                                                                                                 throws InterruptedException, IOException {
-        for (BuildStep publisher: getPublisherList()) {
-            if (!runner.perform(condition, publisher, build, launcher, listener)) {
+        if (getPublisherList().size() <= 0) {
+            return true;
+        } else if (getPublisherList().size() > 1) {
+            return runner.perform(condition, new RunAllBuilder(getPublisherList()), build, launcher, listener);
+        }
+        
+        // the special case for only one publisher.
+        // this allows workarounds when RunAllBuilder causes problems.
+        BuildStep buildstep = getPublisherList().get(0);
+        try {
+            if (!runner.perform(condition, getPublisherList().get(0), build, launcher, listener)) {
+                listener.error(String.format(
+                        "[flexible-publish] %s failed",
+                        FlexiblePublisher.getBuildStepName(buildstep)
+                ));
+                build.setResult(Result.FAILURE);
                 return false;
             }
+        } catch (Exception e) {
```

Comment:
I'm not sure.
Jenkins core doesn't seem handle `InterruptedException` in the special way.

You point the behavior when you abort a build, don't you?
I'll test the behavior when I abort a build when a publisher is running.